### PR TITLE
Templates for android

### PR DESCRIPTION
**Pull request by wmathurin**
Also changes to ios templates: install.sh is now install.js (node js script).

### CODE REVIEW

File: AndroidNativeTemplate/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "AndroidNativeTemplate",
+    "version": "0.0.1",
+    "private": true,
+    "dependencies": {
+        "salesforcemobilesdk-android": "https://github.com/wmathurin/SalesforceMobileSDK-Android.git#templates"
```
Comment (bhariharan) [inline AndroidNativeTemplate/package.json]: Change to `forcedotcom` and fix `URL`.

File: HybridLocal/bootconfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "remoteAccessConsumerKey": "3MVG9Iu66FKeHhINkB1l7xt7kR8czFcCTUhgoA8Ol2Ltf1eYHOU4SqQRSEitYFDUpqRWcoQ2.dBv_a1Dyu5xa",
+    "oauthRedirectURI": "testsfdc:///mobilesdk/detect/oauth/done",
+    "oauthScopes": ["api"],
```
Comment (bhariharan) [inline HybridLocal/bootconfig.json]: Don't we need `web` scope too?

File: HybridLocal/package.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "HybridLocal",
+    "version": "0.0.1",
+    "private": true,
+    "dependencies": {
+        "salesforcemobilesdk-shared": "https://github.com/wmathurin/SalesforceMobileSDK-Shared.git#templates",
```
Comment (bhariharan) [inline HybridLocal/package.json]: Change to `forcedotcom`. Should branch be `templates`?

**Comment by wmathurin** [comment]
NB there are still issues being resolved - but i thought review/feedback could start

File: ReactNativeTemplate/android/app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="account_type">com.salesforce.reactnativetemplat.login</string>
```
Comment (bhariharan) [inline ReactNativeTemplate/android/app/src/main/res/values/strings.xml]: Typo: `template` missing `e` at the end.

File: ReactNativeTemplate/package.json [hunk 5]
```diff
@@ -10,7 +10,8 @@
         "react": "15.3.2",
         "react-native": "0.35.0",
         "salesforcemobilesdk-ios": "https://github.com/forcedotcom/SalesforceMobileSDK-iOS.git#unstable",
-        "react-native-force": "https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative.git#unstable"
+        "react-native-force": "https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative.git#unstable",
+        "salesforcemobilesdk-android": "https://github.com/wmathurin/SalesforceMobileSDK-Android.git#templates"
```
Comment (bhariharan) [inline ReactNativeTemplate/package.json]: Change to `forcedotcom`. Shouldn't branch be `unstable`?

File: ReactNativeTemplate/android/gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#Mon Aug 22 10:53:08 PDT 2016
```
Comment (bhariharan) [inline ReactNativeTemplate/android/gradle/wrapper/gradle-wrapper.properties]: Should we move these common files, i.e., `gradle-wrapper.jar`, `gradle-wrapper.properties`, `gradlew` to a common top level folder under `android` and symlink them here for each app template?

File: ReactNativeTemplate/package.json (hunk 5 above)
Comment (wmathurin) [inline ReactNativeTemplate/package.json]: Yeah once the PR to unstable goes through.

**Comment by wmathurin** [comment]
At this point, what's left is really bug fixes. So I'm going to merge this one in, and then more PRs with the fixes (that will also allow others to join in on the effort).

**Pull request merged**